integrations/astra/tests/test_document_store.py

```diff
@@ -0,0 +1,266 @@
+# SPDX-FileCopyrightText: 2023-present John Doe <jd@example.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+import os
+from typing import List
+
+import pytest
+from haystack import Document
+from haystack.document_stores import DuplicatePolicy, MissingDocumentError
+from haystack.testing.document_store import DocumentStoreBaseTests
+
+from astra_store.document_store import AstraDocumentStore
+
+
+class TestDocumentStore(DocumentStoreBaseTests):
```

Comment:
We should also have at least minimal tests for the retriever and the filters. For example, you could have a look at pinecone: https://github.com/deepset-ai/haystack-core-integrations/blob/main/integrations/pinecone/tests/test_dense_retriever.py and https://github.com/deepset-ai/haystack-core-integrations/blob/main/integrations/pinecone/tests/test_filters.py

Reply:
Can you clarify this point a little bit? @julian-risch

Reply:
We are running the same filter tests as Pinecone. Their test_filters.py inherits tests from FilterDocumentsTest in haystack/haystack/testing/document_store.py, while our test_document_store inherits its tests from DocumentStoreBaseTests which includes the tests from FilterDocumentsTest

Reply:
We will be using the retriever test though